### ISSUE TITLE
Inaccurate cursor positioning on Safari

### ISSUE THREAD

**Issue by shootingfly** [josdejong/jsoneditor#502]
I  can get accurate cursor position on Chrome, Firefox, IE, except safari.  
![dingtalk20180112145504](https://user-images.githubusercontent.com/20180317/34863185-056e5e4e-f7a9-11e7-829a-6fc2f64b787f.png)

To solve the problem, I have to open ``jsoneditor.js`` and delete ``font-family``.
``jsoneditor.js#36892``
```css
	.ace-jsoneditor.ace_editor {\
	font-family: droid sans mono, consolas, monospace, courier new, courier, sans-serif;\
	line-height: 1.3;\
	background-color: #fff;\
	}\
```
Another way is to add style to page.
```css
  .ace-jsoneditor.ace_editor {
    font-family: courier;
  }
```

Is it a bug?

**Comment by josdejong** [comment]
Thanks for reporting. I opened http://jsoneditoronline.org Safari 10.1, which seems to work fine. Does this website work fine for you too?

Can it be that you haven't defined charset on your webpage or something? Like: 

```html
<head>
  <meta http-equiv="Content-Type" content="text/html;charset=utf-8">
  <script type="text/javascript" src="app.min.js" charset="utf-8"></script>
</head>
```

**Comment by shootingfly** [comment]
@josdejong 
Thanks for answer. The website work fine for me. But it is not because charset. I have set it.
If I use an non-monospaced font, it will go wrong.

I haven't install two fonts (droid sans mono and consolas), so Safari will render a kind of monospace font. On Chrome, Courier is my default monospaced font. But Safari will relate to the language of html. Mine is Chinese.
``` html
<html lang="zh-CN"> 
```
When I remove the lang attributes or set it to 'en' or 'jp', all will be ok.
Another way is to install some fonts, such as consolas, droid sans mono.

...so it may because that Safari cannot give me a right monospaced font.

**Comment by josdejong** [comment]
Looks indeed like Ace editor only works with monospaced fonts: https://github.com/ajaxorg/ace/issues/460.

I suppose though that normally there are monospaced fonts installed on Mac and Safari (normally Consolas I think?). At least looks like that's the most logic solution: make sure there is a mono font available.

**Comment by shootingfly** [comment]
Thanks. The ace ssue has been opened for a long time, since 2010. 
Courier is one of monospaced fonts and almost installed in every system.
Can change the order of the font-family in code?
**Old**
``` css
font-family: droid sans mono, consolas, monospace, courier new, courier, sans-serif;\
```
**New**
```css
font-family: droid sans mono, consolas,  courier, courier new,  monospace, sans-serif;\
```

**Comment by josdejong** [comment]
That makes sense, putting the generic "monospace" last. In that case I will add a few more fonts, something like: 

```css
font-family: "dejavu sans mono", "droid sans mono", consolas, monaco, "lucida console", "courier new", courier, monospace, sans-serif;
```

**Comment by josdejong** [comment]
@shootingfly I've just released v5.13.2, which should have reordered font-family, can you check whether this solves your issue?

**Comment by shootingfly** [comment]
@josdejong Yeah! It solves my issue. Thanks.

**Comment by josdejong** [comment]
:+1:

**Comment by danielkrich** [comment]
Hi there! Happens to me as well, on Chrome - version 78.0.3904.108 (Official Build) (64-bit)
I have a general css file with: 
```
* {
      font-family: 'Source Sans Pro', sans-serif;
}
```
And for some reason the json editor is not overriding this font. 
Any ideas? @josdejong 🙏🏼

**Comment by josdejong** [comment]
Ah, that's interesting! Thanks for sharing Daniel. Ace editor has a lot of nested components and simply defines the font-family in the root element, making use of the cascading feature of CSS. Your `* { ... }` rule overrules everything. It's quite an aggressive rule...

We can quite easily fix this by doing:

```css
* {
  font-family: 'Source Sans Pro', sans-serif;
}

.ace_editor * {
    font-family: "dejavu sans mono", "droid sans mono", consolas, monaco, "lucida console", "courier new", courier, monospace, sans-serif;
}
```

I've addressed this in #7c3695d (fix not yet published)

**Comment by josdejong** [comment]
@danielkrich this has been fixed now in `v8.1.0`

**Comment by danielkrich** [comment]
Will check. Thanks a lot @josdejong, highly appreciated! 👏
